index.bs

```diff
@@ -403,107 +410,100 @@ these steps:
 
 <xmp class=idl>
 [Exposed=(Window,Worker), SecureContext]
-interface FileSystemWriter {
+interface FileSystemWritableFileStream {
   Promise<void> write(unsigned long long position, (BufferSource or Blob or USVString) data);
-  WritableStream asWritableStream();
   Promise<void> truncate(unsigned long long size);
   Promise<void> close();
 };
 </xmp>
 
-A {{FileSystemWriter}} has an associated <dfn for=FileSystemWriter>atomic flag</dfn>.
-
-### The {{FileSystemWriter/write()}} method ### {#api-filesystemwriter-write}
-
-<div class="note domintro">
-  : await |writer| . {{FileSystemWriter/write()|write}}(|position|, |data|)
-  :: Writes the content of |data| into the file associated with |writer| at position |position|.
-     If |position| is past the end of the file writing will fail and this method rejects.
-
-     There are no guarantees that the data written to the file using this method is actually
-     reflected in the file on disk until the {{FileSystemWriter/close()}}
-     method is called. Specifically when |writer|'s
-     [=atomic flag=] is set, the user agent will ensure that no changes are written to the actual
-     file until one of those methods is called.
-</div>
+A {{FileSystemWritableFileStream}} object is a {{WritableStream}} object with additional methods to operate
+on a single file on disk.
 
-<div algorithm>
-The <dfn method for=FileSystemWriter>write(|position|, |data|)</dfn> method, when invoked, must run
-these steps:
+The {{FileSystemWritableFileStream}} can be initialized with a |position|. When obtaining a writer through
+{{WritableStream/getWriter()|WritableStream's getWriter()}}, and using the returned
+{{WritableStreamDefaultWriter/write()|WritableStreamDefaultWriter's write()}} method, this position will be
+advanced based on the number of bytes written through this stream object.
 
-1. TODO
+{{WritableStream/getWriter()|WritableStream's getWriter()}} method returns a raw {{WritableStreamDefaultWriter}}.
 
-</div>
+Similarly, when piping a {{FileSystemWritableFileStream}} object, this position is updated with the number
+of bytes that passed through the stream.
 
-### The {{FileSystemWriter/asWritableStream()}} method ### {#api-filesystemwriter-aswritablestream}
+A {{FileSystemWritableFileStream}} has an associated <dfn for=FileSystemWritableFileStream>atomic flag</dfn>.
 
-Issue(19): The functionality described by this method will likely stay, but the exact shape of
-the method is still in flux even more than the rest of this specification.
+### The {{FileSystemWritableFileStream/write()}} method ### {#api-filesystemwritablefilestream-write}
 
 <div class="note domintro">
-  : let |stream| = |writer| . {{FileSystemWriter/asWritableStream()}}
-  :: Returns a {{WritableStream}} that can be used to write data into the file, starting
-     at the beginning of the file.
+  : await |writableFStream| . {{FileSystemWritableFileStream/write()|write}}(|position|, |data|)
```

Comment:
do we want this (and I suppose possibly truncate below as well) to update the position at which future getWriter() calls will start writing?

Reply:
Good question. Given that this takes an explicit `position`, my take on it would be that it wouldn't, but I look forward to the discussion on it.

After calling `write()` with an explicit position, let's consider two possible behaviors:
1. the implicit `position` is changed to the last byte just written
2. the implicit `position` is unchanged

To me, number 2 seems to be the least surprising, because the use of an explicit position option seems to imply eschewing the use of stream-like functionality, perhaps temporarily, for explicit control on the start position.

Reply:
updated `write` to not take `position` as a positional argument.